optuna/distributions.py

```diff
@@ -678,3 +678,76 @@ def _get_single_value(distribution: BaseDistribution) -> Union[int, float, Categ
     elif isinstance(distribution, CategoricalDistribution):
         return distribution.choices[0]
     assert False
+
+
+def _warn_old_distribution_conversion(
```

Comment:
How about a note that this feature will also be removed when the deprecated distributions are removed?

Reply:
It makes sense. Thank you for the suggestion!

Reply:
[0fa53e3](https://github.com/optuna/optuna/pull/3420/commits/0fa53e3179a5c0650f81b59d5fccbac217ae24e2)